rb_ws/src/buggy/scripts/auton/path_planner.py

```diff
@@ -78,69 +93,92 @@ def compute_traj(
+        nominal_slice_to_other_dist = np.abs(nominal_slice_dists - other_dist)
+
+        passing_offsets = self.offset_func(nominal_slice_to_other_dist)
+
+        # compute signed cross-track distance between NAND and nominal
+        nominal_to_other = np.array((other_pose.x, other_pose.y)) - \
+            np.array(self.nominal_traj.get_position_by_index(other_idx))
+
+        # dot product with the unit normal to produce left-positive signed distance
+        other_normal = self.nominal_traj.get_unit_normal_by_index(np.array(other_idx.ravel()))
+        other_cross_track_dist = np.sum(
+            nominal_to_other * other_normal, axis=1)
+
+        self.other_buggy_xtrack_publisher.publish(Float64(other_cross_track_dist))
+
+        # here, use passing offsets to weight NAND's cross track signed distance:
+        # if the sample point is far from SC, the cross track distance doesn't matter
+        # if the sample point is near, we add cross track distance to the offset,
+        # such that the resulting offset is adjusted by position of NAND
+
+        passing_offsets = passing_offsets + \
+            self.activate_other_crosstrack_func(nominal_slice_to_other_dist) * other_cross_track_dist
+
+        # clamp passing offset distances to distance to the curb
+        if not self.left_curb is None:
+            # grab slice of curb correponding to slice of nominal trajectory.
+            curb_idx = self.left_curb.get_closest_index_on_path(self_pose.x, self_pose.y)
+            curb_dist_along = self.left_curb.get_distance_from_index(curb_idx)
+            curb_idx_end = self.left_curb.get_closest_index_on_path(nominal_slice[-1, 0], nominal_slice[-1, 1])
+            curb_dist_along_end = self.left_curb.get_distance_from_index(curb_idx_end)
+            curb_dists = np.linspace(curb_dist_along, curb_dist_along_end, self.RESOLUTION)
+
+            curb_slice = np.empty((self.RESOLUTION, 2))
+            for i in range(self.RESOLUTION):
+                curb_slice[i, :] = np.array(self.left_curb.get_position_by_distance(
+                    curb_dists[i]
+                ))
+
+            # compute distances from the sample points to the curb
+            nominal_slice_to_curb_dist = np.linalg.norm(curb_slice - nominal_slice, axis=1)
+            passing_offsets = np.minimum(passing_offsets, nominal_slice_to_curb_dist - self.CURB_MARGIN)
+
+        # clamp negative passing offsets to zero, since we always pass on the left,
+        # the passing offsets should never pull SC to the right.
+        passing_offsets = np.maximum(0, passing_offsets)
+
+        # shift the nominal slice by passing offsets
+        nominal_normals = self.nominal_traj.get_unit_normal_by_index(
+            self.nominal_traj.get_index_from_distance(nominal_slice_dists)
         )
-        passing_targets = other_future_poses + self.PASSING_OFFSET * future_pose_unit_normal
+        positions = nominal_slice + (passing_offsets[:, None] * nominal_normals)
```

Comment:
is the `None` array slicing here because `passing_offsets` is multi-dimensional?

Reply:
passing offsets is an (N, ) array, but nominal normals is (N, 2). For the brodcast to work, passing offsets needs to be (N, 1), the None array slicing adds a new empty axis and turns passing offsets to (N, 1)